File: 02_recursion_basics/02_03_print_n_to_1/print_n_to_1.cpp
#include <bits/stdc++.h>

using namespace std;

void print_n_to_1(int i, int n)
{
    if (i > n)
    {
        return;
    }

    print_n_to_1(i + 1, n);
    cout << i << endl;
}

int main()
{
    int num = 5;
    print_n_to_1(1, num);
}

File: 02_recursion_basics/02_03_print_n_to_1/PrintNTo1.java
public class PrintNTo1 {
    public static void main(String[] args) {
        int num = 5;
        print_n_to_1(1, num);
    }

    private static void print_n_to_1(int i, int n) {
        if (i > n) {
            return;
        }

        print_n_to_1(i + 1, n);
        System.out.println(i);
    }
}
